### DISCUSSION TITLE
How to get the data for one specific id defined under AcDbRegAppTableRecord

### DISCUSSION THREAD

**Discussion by puppybrown** [mozman/ezdxf#827] [Q&A] [answered]
Hello. I am new with dxf modifying but want to work something out. 

I have a dxf file in which the points are assigned with a special id named 'pnum'. This 'pnum' is defined under AcDbRegAppTableRecord and for each POINT entity one number is assigned to this 'pnum'. I tried to read the value of it under each POINT entity but the appdata class, xdata class didn't work. Get_dxf_attrib is also tried but didn't work. 

I just want to know that is there any way for ezdxf to obtain this 'pnum' for every Point entity? Here are the code from my dxf file.

for defining the 'pnum':
```Text
0
APPID
5
F1
330
1B
100
AcDbSymbolTableRecord
100
AcDbRegAppTableRecord
2
pnum
70
0
```
Here are the code for Point Entity that assigned with 'pnum':
```Text
0
POINT
5
FF
330
2
100
AcDbEntity
8
testpoint
100
AcDbPoint
10
1065.985740814411
20
1076.680660975082
30
0.0
1001
pnum
1000
2631'
```

Thank you. I would really appreciate your help.

**Comment by mozman** [comment, marked as answer]
The extended data (XDATA) of an entity for a specific APPID can be retrieved using the `get_xdata()` method of any DXF entity:

```Python
from ezdxf.entities import Point

point: Point = Point.from_text("""0
POINT
5
FF
330
2
100
AcDbEntity
8
testpoint
100
AcDbPoint
10
1065.985740814411
20
1076.680660975082
30
0.0
1001
pnum
1000
2631'
""")

if point.has_xdata("pnum"):
    xdata = point.get_xdata("pnum")
    print(xdata)
    print(f"The id of point {str(point)} is : {xdata[0].value}")
```
Output:
```Text
[DXFTag(1000, "2631'")]
The id of point POINT(#FF) is : 2631'
```

The  `get_xdata()` method returns a Python list of `DXFTag` objects, where each tag has a `code` and a `value` attribute. The group code 1000 defines a string tag.

If you want to know more, read the [documentation](https://ezdxf.mozman.at/docs/dxfinternals/dxftags.html#extended-data) about XDATA in the DXF internals section.

**Reply by puppybrown** [reply]
Hello Mozman, thank you very much for your help! It really did solve my problem.
